extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/HibernateSearchElasticsearchBuildTimeConfigPersistenceUnit.java

```diff
@@ -214,4 +219,28 @@ interface CoordinationConfig {
         Optional<String> strategy();
     }
 
+    @ConfigGroup
+    interface MappingConfig {
+        /**
+         * One or more xref:hibernate-search-orm-elasticsearch.adoc#bean-reference-note-anchor[bean references]
+         * to the component(s) used to configure Hibernate Search mapping.
+         *
+         * The referenced beans must implement `HibernateOrmSearchMappingConfigurer`.
+         *
+         * See xref:hibernate-search-orm-elasticsearch.adoc#programmatic-mapping[Programmatic mapping] for an example
+         * on how mapping configurers can be used to apply programmatic mappings.
+         *
+         * [NOTE]
+         * ====
+         * Instead of setting this configuration property,
+         * you can simply annotate your custom `HibernateOrmSearchMappingConfigurer` implementations with `@SearchExtension`
+         * and leave the configuration property unset: Hibernate Search will use the annotated implementation automatically.
+         * Any `@SearchExtension` annotated configurers will be appended to the ones provided by this property.
```

Comment:
Ok if you feel strongly about this, but I think it would be safer (simpler) to just have the property override CDI config.

I feel like mixing CDI with properties could lead to complex behavior, and there's already an example right here... usually in this extension, properties override CDI config, but here you apply CDI configurers after the property configurers, so it's the opposite: CDI configurers can override property configurers.

Besides, if people want to merge CDI configurers with property configurers, they can totally refer to CDI-defined configurers in the property (with `bean:someName`).

Reply:
At the time I thought that the users wouldn't need to override the extension and appending all of them made sense. But to your point -- let's be consistent in behaviour 😃